### PR TITLE
Add Service, SaveService and CallMethodCallbacks

### PR DESCRIPTION
**Pull request by nweathersbee**
This brings in `MuchRails::Service`, `MuchRails::SaveService` and
`MuchRails::CallMethodCallbacks`. Descriptions for each mix-in are
included in the mix-in itself.

### CODE REVIEW

File: test/unit/lib/much-rails/call_method_callbacks_tests.rb [hunk 1]
```diff
@@ -0,0 +1,187 @@
+require "assert"
+
+require "much-rails/call_method_callbacks"
```
Comment (kellyredding) [inline test/unit/lib/much-rails/call_method_callbacks_tests.rb]: I may have told you wrong earlier, but the convention for requires in tests is:

```ruby
require "assert"
require "file_for_unit"

require "other_test_dependency1"
require "other_test_dependency2"

# ...
```
So can you remove the empty lines between `require "assert"` and the test unit file require? And add a empty line between those and any other test dependency requires please? 🙏🏻

File: test/unit/lib/much-rails/call_method_callbacks_tests.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    should "be configured as expected" do
+      assert_that(subject).includes(MuchRails::CallMethod)
+      assert_that(subject).includes(MuchRails::Config)
+    end
+
+    should "know its attributes" do
+      # before_callbacks
+      assert_that(
+        subject.much_rails_call_callbacks_config.before_callbacks.size)
+        .equals(0)
```
Comment (kellyredding) [inline test/unit/lib/much-rails/call_method_callbacks_tests.rb]: Style thing: we are going to wrap closing `)` or `}` to the next line in general, so this becomes:

```suggestion
      assert_that(
        subject.much_rails_call_callbacks_config.before_callbacks.size
      ).equals(0)
```
The `.equals doesn't go on its own line b/c it isn't chained. Make sense? Update others please 🙏🏻 .

File: test/unit/lib/much-rails/call_method_callbacks_tests.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            "before 1",
+            "before 2",
+            "start around 1",
+            "start around 2",
+            "on call",
+            "end around 2",
+            "end around 1",
+            "after 1",
+            "after 2"
+          ])
```
Comment (kellyredding) [inline test/unit/lib/much-rails/call_method_callbacks_tests.rb]: wrap closing `)`

```suggestion
        .equals([
          "before 1",
          "before 2",
          "start around 1",
          "start around 2",
          "on call",
          "end around 2",
          "end around 1",
          "after 1",
          "after 2"
        ])
```

**Review by kellyredding — APPROVED** [review]
@nweathersbee thanks! approving to not block - a few style things I'd like to clean up. If there are other cleanups needed b/c of my comments and you'd prefer to do the cleanups in mass in a future commit/PR, no worries.

**Review by jcredding — APPROVED** [review]

**Pull request merged**